views/contexts/model.go

```diff
@@ -179,6 +234,12 @@ func (m *Model) MoveCursor(delta int) {
 	if m.cursor >= len(m.contexts) {
 		m.cursor = len(m.contexts) - 1
 	}
+	// Mirror to internal FilterableList cursor and ensure visible
+	if m.List.Cursor != m.cursor {
+		m.List.Cursor = m.cursor
+		// ApplyFilter will keep the cursor in-bounds and update viewport offset
+		m.List.ApplyFilter()
+	}
```

Comment:
Why do we even have separate cursors? This makes everything so much more complicated, and I cannot see the benefit: https://github.com/mosonyi/swarmcli/issues/149